cmd/gateway-azure.go

```diff
@@ -512,7 +517,13 @@ func (a *azureObjects) GetObjectInfo(bucket, object string) (objInfo ObjectInfo,
 // PutObject - Create a new blob with the incoming data,
 // uses Azure equivalent CreateBlockBlobFromReader.
 func (a *azureObjects) PutObject(bucket, object string, data *HashReader, metadata map[string]string) (objInfo ObjectInfo, err error) {
+	if metadata == nil {
+		metadata = make(map[string]string)
```

Comment:
Neither deleting from a nil map or ranging over a nil map are errors - so it is not clear why this allocation is needed.

Reply:
Yes i know, old code not removing in this PR.